### ISSUE TITLE
Show email address and add drop down to select a different one when squash merging

### ISSUE THREAD

**Issue by snoack** [Reviewable/Reviewable#1014]
Implement an email address selector similar to the one [GitHub added](https://github.blog/changelog/2022-09-15-git-commit-author-shown-when-squash-merging-a-pull-request/) in the merge options component.

**Comment by snoack** [comment]
Should we remember the choice and default to it in future reviews of that user?

**Comment by pkaminski** [comment]
Yes, we should probably remember a per-repo (or per-organization?) implicit default.

Incidentally, this will require us to switch from the REST API to the [GraphQL one](https://docs.github.com/en/graphql/reference/mutations#mergepullrequest) for merging, as the new parameter hasn't been added to the former.

**Comment by snoack** [comment]
I just noticed is that GitHub defaults to the email address used in the commits to be squashed. If Reviewable would use the email address form the commits rather than the default email address that would already be a big improvement. I'm not even sure, if I'd even need a drop down then.

**Comment by pkaminski** [comment]
We just use the sole GitHub REST API to trigger the merge, and since it doesn't even allow specifying an email address you'd think they'd apply their defaulting logic automatically.  Maybe their GraphQL API does better?  Might be worth checking, and we'd need to switch to it anyway to specify an email.

If we can't leverage GitHub's logic then I agree we should implement it ourselves.  Of course, it gets tricky if the commits use multiple email addresses, or email addresses that aren't included in the user's verified ones.
